lib/render/RayCaster.cpp

```diff
@@ -232,27 +267,31 @@ RayCaster::UserCoordinates::GetCurrentGrid( const RayCasterParams* params,
     {
         MyBase::SetErrMsg("UserCoordinates::GetCurrentGrid() isn't on a StructuredGrid; "
                           "the behavior is undefined in this case.");
+        return 1;
+    }
+    else
+    {
+        *gridpp = grid;
+        return 0;
     }
-    return grid;
 }
 
-bool RayCaster::UserCoordinates::IsUpToDate( const RayCasterParams* params,  
-                                                   DataMgr*         dataMgr ) const
+int  RayCaster::UserCoordinates::IsMetadataUpToDate( const RayCasterParams* params,  
+                                                           DataMgr*         dataMgr ) const
 {
     if( ( myCurrentTimeStep  != params->GetCurrentTimestep()  )  ||
         ( myVariableName     != params->GetVariableName()     )  ||
         ( myRefinementLevel  != params->GetRefinementLevel()  )  ||
         ( myCompressionLevel != params->GetCompressionLevel() )     )
     {
-        return false;
+        return 1;
     }
 
     // compare grid boundaries and dimensions
-    StructuredGrid* grid = this->GetCurrentGrid( params, dataMgr );
-    if( !grid )
+    StructuredGrid* grid  = nullptr;
+    if( this->GetCurrentGrid( params, dataMgr, &grid ) != 0 )
```

Comment:
I am not sure how expensive getting the grid is however you should be able to get the extends from the params directly without requesting a new grid.

Reply:
The params keeps boundaries in ASCII, which is most likely a slightly different value from the intended floating point value. Thus, comparing against values returned by params will most likely result in `false`, even though they're supposed to be `true`. 
Note this only affects floating point values return by params.
@clyne  is this correct?

Reply:
Not correct, per our discussion. If the value cached in RayCaster.cpp is set from a value obtained from the Params subsequent comparisons will only be unequal if the value in the Params is explicitly changed.

Reply:
While it is true the mapping between floating point and real numbers is not one to one, the code that converts between the two is deterministic